src/index.js

```diff
@@ -112,10 +112,9 @@ module.exports.ServerController = function (state) {
     accessors: {
       get: function (path) {
         path = typeof path === 'string' ? path.split('.') : path
-        while (path.length) {
-          state = state[path.shift()]
-        }
-        return state
+        return path.reduce(function (currentState, key) {
```

Comment:
Hm... yeah, I think you are right @unity-haiwei. It should return "undefined". So:

```js
  return path.reduce(function (currentState, key) {
      return currentState === undefined || currentState === null ? undefined : currentState[key]
   }, state)
```

Now it returns undefined if the path does not exist or you try to access a key on `null`.